### PR TITLE
Allow multiple SetData calls on AppDomain

### PR DESCRIPTION
**Pull request by AlexGhiondea**
/cc @sepidehMS

### CODE REVIEW

**Comment by sepidehMS** [comment]
Thanks for doing this @AlexGhiondea

**Comment by jkotas** [comment]
You may also want to get rid of `AppDomainSetup.Locate` and the special casing from GetData, so that the following works same way as on desktop:

```
static void Main() {
    AppDomain.CurrentDomain.SetData("APPBASE", "c:\\in the middle of nowhere");
    Console.WriteLine(AppDomain.CurrentDomain.GetData("APPBASE"));
}
```

It should print `c:\in the middle of nowhere`.

**Comment by AlexGhiondea** [comment]
@dotnet-bot test Tizen armel Cross Debug Build

File: src/mscorlib/src/System/AppDomainSetup.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ internal sealed class AppDomainSetup
         internal enum LoaderInformation
         {
             // If you add a new value, add the corresponding property
-            // to AppDomain.GetData() and SetData()'s switch statements,
+            // to AppDomain.pa() and SetData()'s switch statements,
```
Comment (justinvp) [inline src/mscorlib/src/System/AppDomainSetup.cs]: pa?

Comment (AlexGhiondea) [inline src/mscorlib/src/System/AppDomainSetup.cs]: *sigh*

Comment (AlexGhiondea) [inline src/mscorlib/src/System/AppDomainSetup.cs]: fixed.

Comment (AlexGhiondea) [inline src/mscorlib/src/System/AppDomainSetup.cs]: @justinvp good catch!

File: src/mscorlib/src/System/AppDomain.cs [hunk 2]
```diff
@@ -590,37 +579,18 @@ public Object GetData(string name)
...
-                    lock (((ICollection)LocalStore).SyncRoot)
-                    {
-                        LocalStore.TryGetValue(name, out data);
-                    }
-                    if (data == null)
-                        return null;
-                    return data;
-                }
-            }
+            if (name.Equals(AppDomainSetup.LoaderOptimizationKey))
```
Comment (jkotas) [inline src/mscorlib/src/System/AppDomain.cs]: This should be removed as well. (LoaderOptimization is not used for anything in CoreCLR.)

**Pull request merged**